### PR TITLE
Media loading stability improvements

### PR DESCRIPTION
**Pull request by francinum**
## Changelog

:cl:
fix: Media System Validation is less likely to die unexpectedly.
add: The Meta Bar now has a jukebox
add: And it can be used by non-bartenders.
/:cl:

### CODE REVIEW

File: code/controllers/subsystem/media.dm [hunk 1]
```diff
@@ -115,16 +84,87 @@ SUBSYSTEM_DEF(media)
...
+		spawn(0)
+			UNTIL(SSmedia.initialized)
+			message_admins("MEDIA: At least 1 Media JSON is invalid. Please check SSMedia.errored_files or config_error.log")
+	return
+
+/// Run media validation checks. Returns null on success, or a log_load_failure compatible tuple-list on failure.
+/datum/controller/subsystem/media/proc/validate_media(json_data, jd_full_filepath, jd_tag_cache)
+	if(!json_data || !jd_full_filepath || !jd_tag_cache)
+		stack_trace("BAD CALLING ARGUMENTS TO VALIDATE_MEDIA")
+		return list("ERR_FATAL", "Record validation was called with bad arguments: [json_data || "#FALSY_DATA"], [jd_full_filepath || "#FALSY_DATA"], [english_list(jd_tag_cache, nothing_text = "#FALSY_DATA")]")
```
Comment (Kapu1178) [inline code/controllers/subsystem/media.dm:120]: can you use more to the point errors, errors are in english we arent writing an operating system

File: code/controllers/subsystem/media.dm [hunk 2]
```diff
@@ -28,15 +30,28 @@ SUBSYSTEM_DEF(media)
 		"mp3" = TRUE, //MPeg Layer 3 Container (And usually, Codec.)
 	)
 
+#define MEDIA_LOAD_FAILED -1
```
Comment (Kapu1178) [inline code/controllers/subsystem/media.dm:33]: why not just use 0

**Review by Kapu1178 — CHANGES_REQUESTED** [review]

**Comment by Kapu1178** [comment]
@francinum

File: code/controllers/subsystem/media.dm (hunk 2 above)
Comment (francinum) [reply code/controllers/subsystem/media.dm:33]: Crash catching. `-1` means we at least returned under control, `null/0` means we returned without control, as in the json file is so bad it killed the proc.

File: code/controllers/subsystem/media.dm (hunk 1 above)
Comment (francinum) [reply code/controllers/subsystem/media.dm:120]: You should never get this particular error *EVER* unless something is so bad you *should* be getting a developer.

If you have a problem with any of the more general errors, let me know.

**Pull request merged**